lms/static/scripts/frontend_apps/utils/date.ts

```diff
@@ -1,12 +1,60 @@
 /**
- * Formats a date as `YYYY-MM-DD hh:mm`, using 24h and system timezone.
+ * Map of stringified `DateTimeFormatOptions` to cached `DateTimeFormat` instances.
  */
-export function formatDateTime(date: Date): string {
-  const year = date.getFullYear();
-  const month = `${date.getMonth() + 1}`.padStart(2, '0');
-  const day = `${date.getDate()}`.padStart(2, '0');
-  const hours = `${date.getHours()}`.padStart(2, '0');
-  const minutes = `${date.getMinutes()}`.padStart(2, '0');
+let formatters = new Map<string, Intl.DateTimeFormat>();
 
-  return `${year}-${month}-${day} ${hours}:${minutes}`;
+/**
+ * Clears the cache of formatters.
+ */
+export function clearFormatters() {
+  formatters = new Map<string, Intl.DateTimeFormat>();
+}
+
+type IntlType = typeof window.Intl;
+
+/**
+ * Return date string formatted with `options`.
+ *
+ * This is a caching wrapper for `Intl.DateTimeFormat.format`, useful because
+ * constructing a `DateTimeFormat` is expensive.
+ *
+ * @param Intl - Test seam. JS `Intl` API implementation.
+ */
+function format(
+  date: Date,
+  options: Intl.DateTimeFormatOptions,
+  /* istanbul ignore next */
+  Intl: IntlType = window.Intl,
+): string {
+  const key = JSON.stringify(options);
+  let formatter = formatters.get(key);
+  if (!formatter) {
+    formatter = new Intl.DateTimeFormat(undefined, options);
+    formatters.set(key, formatter);
+  }
+  return formatter.format(date);
+}
+
+/**
+ * Formats a date as an absolute string in a human-readable format.
+ *
+ * The exact format will vary depending on the locale, but the verbosity will
+ * be consistent across locales. In en-US for example this will look like:
+ *
+ *  "Dec 17, 2017, 10:00 AM"
+ *
+ * @param Intl - Test seam. JS `Intl` API implementation.
+ */
+export function formatDate(date: Date, Intl?: IntlType): string {
```

Comment:
I preferred the `formatDateTime` name we used before, because it made it clearer that the output included both date and time components.